### PR TITLE
test cases added for address book, category, customer service layer

### PR DESCRIPTION
**Pull request by mariappan-sivakumar-zuci**
Hi Jithu,
I have added testcases for 3service layer we met with a error on Monday the whole day is spend on that error after that. We are surf on the internet to study how to write unit test cases. so why two days we didn't make any pull request
If you found any correction please let me know.

### CODE REVIEW

**Comment by mariappan-sivakumar-zuci** [comment]
Hi Jithu,
I have added category and subcategory test In this commit I starting working on writing testcase for Product service layer but I didn't complete that.

File: src/main/java/com/app/shopping/ecommerce/entity/Order.java [hunk 1]
```diff
@@ -26,5 +26,5 @@ public class Order {
     private Date placedOn;
     private Date deliveryOn;
     private String status;
-    private List<Product> products;
+//    private List<Product> products;
```
Comment (jithuzuci) [inline src/main/java/com/app/shopping/ecommerce/entity/Order.java:29]: Why is it commented on here? If not needed can you please remove it?

**Review by jithuzuci — CHANGES_REQUESTED** [review]
I have made a change request, please address it before merging.

**Comment by jithuzuci** [comment]
Also, as mentioned in the previous PR comment  https://github.com/mariappan-sivakumar-zuci/E-Commerce/pull/46#pullrequestreview-1712724428, Please try to make smaller PR's otherwise it's hard to review. You can try something like a PR per service or methods.
Also, you can split the work within your group and raise separate PRs?

File: src/main/java/com/app/shopping/ecommerce/services/SupplierService.java [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
-import com.app.shopping.ecommerce.entity.Supplier;
 import com.app.shopping.ecommerce.payload.SupplierDto;
 import com.app.shopping.ecommerce.payload.SupplierReg;
 
 import java.util.List;
-import java.util.Optional;
 
 public interface SupplierService {
-    SupplierReg RegistrationSupplier(SupplierReg supplierreg);
+    SupplierReg registrationSupplier(SupplierReg supplierreg);
```
[the remark this answers is not in the document]
Comment (Jaya-Anand-Natesan) [reply src/main/java/com/app/shopping/ecommerce/services/SupplierService.java:9]: Okay Sure 
I will change it.
Thank you.

File: src/main/java/com/app/shopping/ecommerce/entity/Order.java (hunk 1 above)
Comment (mariappan-sivakumar-zuci) [reply src/main/java/com/app/shopping/ecommerce/entity/Order.java:29]: We are hold the process in order entity. with the list of product we face difficulties in the database section with out the annotations that's why we commented it after some time we will un-commend it and work on it

**Review by jithuzuci — APPROVED** [review]
Thanks for making the changes. It looks good to me.

**Comment by mariappan-sivakumar-zuci** [comment]
Ok Jithu

**Comment by Jaya-Anand-Natesan** [comment]
Thank you Jithu. It means a lot..

**Pull request merged**